### DISCUSSION TITLE
Support Script In GraphQL?

### DISCUSSION THREAD

**Discussion by 40lsgy1** [graphql-java/graphql-java#2772] [General]
In my case,  I want to construct the output value conditional on expression。I can use @include directive，but the if parameter just accept boolean，maybe script can be more flexible？such as：
```
query Hero($episode: Episode, $friendKind: Boolean!) {
  hero(episode: $episode) {
    name
    friends @include(if: $friendKind == 'DOG') {
      name
    }
  }
}
```

**Comment by dugenkui03** [comment]
@40lsgy1 I don't think `@include` will support script, since the implementation of `@include` in graphql-java obey the [sec--include](https://spec.graphql.org/October2021/#sec--include)

That is the [graphql calculator way](https://github.com/graphql-calculator/graphql-calculator#%E6%8E%A7%E5%88%B6%E6%B5%81), you can custom directive which satisfy the ' include by script':
```graphql
query skipBy_case01($userId:Int){
    consumer{
        userInfo(userId: $userId)
        # the userInfo field would not be queried if 'userId>100' is true
        @skipBy(predicate: "userId>100")
        {
            userId
            name
        }
    }
}
```

**Reply by 40lsgy1** [reply]
well, it works, thx bro.

**Comment by bbakerman** [comment]
As kinda cool as it first might appear- the graphql-java engine wont be supporting "scripting" in queries - not unless it started turning up as a graphql spec defined thing
